### DISCUSSION TITLE
Autoindent isn't automatically applied when creating a new line using Lua API

### DISCUSSION THREAD

**Discussion by anthony-S93** [neovim/neovim#27329] [Q&A] [answered]
I have a custom keymap that allows me to use the `o` command in `NORMAL` mode while inside `INSERT` mode:
```lua
vim.keymap.set("i", "<C-j>", function() vim.cmd(":normal o") end, opts)
```
The problem is that I can't figure out how to also trigger the autoindent feature when I use the custom keymap:

https://github.com/neovim/neovim/assets/69449791/d37d6c82-e72c-4672-a067-c00c3148bcfb

When using `o` from `NORMAL` mode:

https://github.com/neovim/neovim/assets/69449791/3a25e81e-994d-4202-8cb1-f95b098a1b47

**Comment by zeertzjq** [comment, marked as answer]
You can use this:
```lua
vim.keymap.set("i", "<C-J>", "<C-O>o", opts)
```

**Reply by anthony-S93** [reply]
This worked. Thanks.  But why didn't `vim.cmd()` work?

**Reply by zeertzjq** [reply]
See the documentation for `:normal`:
```
			{commands} should be a complete command.  If
			{commands} does not finish a command, the last one
			will be aborted as if <Esc> or <C-C> was typed.
			This implies that an insert command must be completed
```
